### ISSUE TITLE
Incorrect substitution for Style/HashExcept

### ISSUE THREAD

**Issue by tobyaw** [rubocop/rubocop#10031]
I have code:

```
{}.reject { |_k, v| v.count.eql? 1 }
```

and the Style/HashExcept rule changes this to invalid and non-equivalent code:

```
{}.except(v.count)
```

## Expected behavior

The Style/HashExcept rule should not trigger.

## Actual behavior

```
% rubocop --debug
For /Users/tobyaw/Desktop/test: configuration from /Users/tobyaw/Desktop/test/.rubocop.yml
Default configuration from /usr/local/lib/ruby/gems/3.0.0/gems/rubocop-1.19.1/config/default.yml
Use parallel by default.
Running parallel inspection
Inspecting 1 file
Scanning /Users/tobyaw/Desktop/test/test.rb
Loading cache from /Users/tobyaw/.cache/rubocop_cache/05049447917d12ccd9963972b8d2f31cd20e5928/6d7a3b621ca1730e04accd938619e4bdab66cfb1/c47fd0571652f491f0955dba32cb5933e95d6a5f
C

Offenses:

test.rb:4:4: C: [Correctable] Style/HashExcept: Use except(v.count) instead.
{}.reject { |_k, v| v.count.eql? 1 }
   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

1 file inspected, 1 offense detected, 1 offense auto-correctable
Finished in 0.19939299998804927 seconds
```

## Steps to reproduce the problem

Sample code:

```
#!/usr/bin/env ruby
# frozen_string_literal: true

{}.reject { |_k, v| v.count.eql? 1 }
```

Rubocop config:

```
AllCops:
  TargetRubyVersion: 3.0
  NewCops: enable
```

## RuboCop version

```
% rubocop -V
1.19.1 (using Parser 3.0.2.0, rubocop-ast 1.10.0, running on ruby 3.0.2 x86_64-darwin20)
```
